File: app/src/main/java/com/cookandroid/e_eum/MainActivity.java
package com.cookandroid.e_eum;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.hardware.Camera;
import android.graphics.PixelFormat;
import android.os.Bundle;
import android.view.SurfaceHolder;
import android.view.SurfaceView;
import android.widget.Button;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    Camera camera;
    Button btn_test_start,btn_test_end;
    SurfaceView sfv_test1;
    SurfaceHolder sfv_holder;
    boolean previewing = false;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn_test_start = findViewById(R.id.btn_test_start);
        btn_test_end = findViewById(R.id.btn_test_end);
        sfv_test1 = findViewById(R.id.sfv_test1);

        getWindow().setFormat(PixelFormat.UNKNOWN);
        sfv_test1 = (SurfaceView)findViewById(R.id.sfv_test1);
        sfv_holder = sfv_test1.getHolder();
//        sfv_holder.addCallback(this);
        sfv_holder.setType(SurfaceHolder.SURFACE_TYPE_PUSH_BUFFERS);
    }
}

File: app/src/main/java/com/cookandroid/e_eum/MainActivity.kt
package com.cookandroid.e_eum

import androidx.appcompat.app.AppCompatActivity
import android.view.SurfaceView
import android.view.SurfaceHolder
import android.os.Bundle
import com.cookandroid.e_eum.R
import android.graphics.PixelFormat
import android.hardware.Camera
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    var camera: Camera? = null
    var btn_test_start: Button? = null
    var btn_test_end: Button? = null
    var sfv_test1: SurfaceView? = null
    var sfv_holder: SurfaceHolder? = null
    var previewing = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_test_start = findViewById(R.id.btn_test_start)
        btn_test_end = findViewById(R.id.btn_test_end)
        sfv_test1 = findViewById(R.id.sfv_test1)
        window.setFormat(PixelFormat.UNKNOWN)
        sfv_test1 = findViewById<View>(R.id.sfv_test1) as SurfaceView
        sfv_holder = sfv_test1!!.holder
    }
}